### ISSUE TITLE
Yauzl throws uncatchable error

### ISSUE THREAD

**Issue by PanMan** [thejoshwolfe/yauzl#65]
yauzl throws this error that I seem to be unable to catch or prevent:
```
 events.js:183
       throw er; // Unhandled 'error' event
      ^
 Error: invalid characters in fileName: some name\, de something something 5.6.vcf
     at /app/node_modules/yauzl/index.js:417:70
     at /app/node_modules/yauzl/index.js:622:5
     at /app/node_modules/fd-slicer/index.js:32:7
     at FSReqWrap.wrapper [as oncomplete] (fs.js:658:17)
```
Am I doing something wrong? This started happening a few weeks ago, on very few files. Sanitizing file names doesnt seem to solve it.

**Comment by thejoshwolfe** [comment]
It looks like you're not listening for the error even on the zipfile object. See https://github.com/thejoshwolfe/yauzl/blob/master/README.md#how-to-avoid-crashing

**Comment by PanMan** [comment]
Thanks, missed that part, added some code and will do some more testing...

**Comment by PanMan** [comment]
You are right, in that I had to listen to the error event.
```
Got an Zip Err Error: invalid characters in fileName: some name\, de something 3e verd. kamer 5.6.vcf
2017-11-17T14:46:54.977807+00:00 app[worker.1]:     at /app/node_modules/yauzl/index.js:417:70
2017-11-17T14:46:54.977808+00:00 app[worker.1]:     at /app/node_modules/yauzl/index.js:622:5
2017-11-17T14:46:54.977808+00:00 app[worker.1]:     at /app/node_modules/fd-slicer/index.js:32:7
2017-11-17T14:46:54.977809+00:00 app[worker.1]:     at FSReqWrap.wrapper [as oncomplete] (fs.js:658:17)
```
However, how do I prevent this? I don't even mind skipping the file (its not a file I need), can I process the file further?

**Comment by andrewrk** [comment]
Did you read the documentation?

**Comment by thejoshwolfe** [comment]
This issue has come up before in #54. A workaround for recovering from zipfiles corrupted in this way isn't documented in the readme, but maybe it should be.

**Comment by thejoshwolfe** [comment]
@PanMan can you tell what client is creating zipfiles with `\` characters in file names?

**Comment by thejoshwolfe** [comment]
Please reopen if you have any other questions.

**Comment by PanMan** [comment]
Sorry, I missed this notification. We use Yauzl to process chat archives created by WhatsApp. It hasn't happened often, but did happen a few times.

**Comment by thejoshwolfe** [comment]
I am interested in reproducing the backslash filename problem. I have WhatsApp for android, and i've been looking into creating a chat backup of conversations.

Is there any advice you can give me to help me reproduce this issue? There's no rush. I may not make any substantial progress on this issue for a few weeks.

**Comment by thejoshwolfe** [comment]
Backslashes are now silently converted to forward slashes as of https://github.com/thejoshwolfe/yauzl/commit/6a9e652e3fee41938bb1fb436f363b3ffc7e0b0f published in yauzl 2.10.0.
